src/main/java/hudson/remoting/RemoteClassLoader.java

```diff
@@ -449,38 +507,75 @@ public URL findResource(String name) {
             return url;
         }
 
+        boolean interrupted = false;
         try {
-            if (resourceMap.containsKey(name)) {
-                URLish f = resourceMap.get(name);
-                if (f == null) {
-                    return null;    // no such resource
-                }
-                URL u = f.toURL();
-                if (u != null) {
-                    return u;
-                }
-            }
+            for (int tries = 0; tries < MAX_RETRIES; tries++) {
+                try {
+                    if (resourceMap.containsKey(name)) {
+                        URLish f = resourceMap.get(name);
+                        if (f == null) {
+                            return null;    // no such resource
+                        }
+                        URL u = f.toURL();
+                        if (u != null) {
+                            return u;
+                        }
+                    }
 
-            long startTime = System.nanoTime();
+                    invokeResourceLoadTestingHookIfNeeded();
 
-            ResourceFile r = proxy.getResource2(name);
-            ResourceImageRef image = null;
-            if (r != null) {
-                image = r.image;
-            }
+                    long startTime = System.nanoTime();
 
-            channel.resourceLoadingTime.addAndGet(System.nanoTime() - startTime);
-            channel.resourceLoadingCount.incrementAndGet();
-            if (image == null) {
-                resourceMap.put(name, null);
-                return null;
+                    ResourceFile r = proxy.getResource2(name);
+                    ResourceImageRef image = null;
+                    if (r != null) {
+                        image = r.image;
+                    }
+
+                    channel.resourceLoadingTime.addAndGet(System.nanoTime() - startTime);
+                    channel.resourceLoadingCount.incrementAndGet();
+                    if (image == null) {
+                        resourceMap.put(name, null);
+                        return null;
+                    }
+
+                    URLish res = image.resolveURL(channel, name).get();
+                    resourceMap.put(name, res);
+                    return res.toURL();
+                } catch (IOException | InterruptedException | ExecutionException e) {
```

Comment:
move `InterruptedException` to the next catch statement, so that it is handled by the retry logic

Reply:
This makes me reconsider the previous change to standardize the exceptions through a single method. Was there some reason InterruptedException was handled the way it was? -- But, I don't have much confidence that the previous exception handling was correct or very intentional. I still think the standardization is probably better.